src/main/java/com/eighthlight/fabrial/utils/Result.java

```diff
@@ -88,6 +88,10 @@ public V orElseThrow() throws E {
     return Optional.ofNullable(value).orElseThrow(() -> error);
   }
 
+  public <R extends E> V orElseThrow(Function<E, R> exceptionMapper) throws R {
+    return Optional.ofNullable(value).orElseThrow(() -> exceptionMapper.apply(error));
+  }
+
```

Comment:
I see the compiler is sending out some warning about unchecked casts in this file - if you're sure these are safe (or at least unbroken in this codebase), you could suppress the warnings with `@SuppressWarnings("unchecked")`.

#nonblocker, not added by this PR.

Reply:
I don't think there's a way around the casts here—at least based on my reading of similar libraries (see https://github.com/aol/cyclops)